### ISSUE TITLE
In inference, confidence scores differ when changing batch size

### ISSUE THREAD

**Issue by mo-soliman** [ultralytics/yolov5#8107]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Detection

### Bug

When using torch.hub.load('ultralytics/yolov5', 'yolov5s')
With a batch size = 1, image = "'https://ultralytics.com/images/bus.jpg', the model gives this result:
![bs1](https://user-images.githubusercontent.com/60558152/172063490-cb79a009-63db-425c-8405-515a02a0c284.png)

However when using batch size = 2, the same image "'https://ultralytics.com/images/bus.jpg' gives different confidence scores:
![bs2](https://user-images.githubusercontent.com/60558152/172063553-ff75767e-8566-40f5-b3ba-2629232c57e3.png)

Note: The second image when batch size = 2 is 'https://ultralytics.com/images/zidane.jpg'

Is this normal? What is the cause of this behavior?

### Environment

YOLOv5 🚀 2022-5-30 Python-3.8.13 torch-1.11.0+cu102 CUDA:0 (Tesla K80, 11441MiB)

OS: Ubuntu 18.04.5 LTS

Python-3.8.13

### Minimal Reproducible Example

import torch

model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  

### \# Batch size = 1
img = 'https://ultralytics.com/images/bus.jpg' 
results = model(img)
results.pandas().xyxy[0] 

### \# Batch size = 2
img = ['https://ultralytics.com/images/bus.jpg', 'https://ultralytics.com/images/zidane.jpg']
results = model(img)
results.pandas().xyxy[0]  

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by mo-soliman** [comment]
After a bit of debugging, I believe this is normal, batch padding varies according to the images within the batch, hence results varies.

**Comment by glenn-jocher** [comment]
@mo-soliman not a bug, normal as padding is required to change.
